scripts/release.ts

```diff
@@ -0,0 +1,229 @@
+import { exec as _exec, spawn } from 'child_process';
+import { promisify } from 'util';
+import yargs, { ArgumentsCamelCase, Argv } from 'yargs';
+import { logger, setLogLevel, LOG_LEVEL } from '@hubspot/local-dev-lib/logger';
+import semver from 'semver';
+
+import { name as packageName, version as localVersion } from '../package.json';
+import { EXIT_CODES } from '../lib/enums/exitCodes';
+import { confirmPrompt } from '../lib/prompts/promptUtils';
+import { build } from './build';
+
+// TODO: Change to import when this is converted to TS
+const { uiLink, uiLine } = require('../lib/ui');
+
+const exec = promisify(_exec);
+
+const MAIN_BRANCH = 'main';
+
+const TAG = {
+  LATEST: 'latest',
+  NEXT: 'next',
+  EXPERIMENTAL: 'experimental',
+} as const;
+
+const INCREMENT = {
+  PATCH: 'patch',
+  MINOR: 'minor',
+  MAJOR: 'major',
+  PRERELEASE: 'prerelease',
+} as const;
+
+const VERSION_INCREMENT_OPTIONS = [
+  INCREMENT.PATCH,
+  INCREMENT.MINOR,
+  INCREMENT.MAJOR,
+  INCREMENT.PRERELEASE,
+] as const;
+const TAG_OPTIONS = [TAG.LATEST, TAG.NEXT, TAG.EXPERIMENTAL] as const;
+
+type ReleaseArguments = {
+  versionIncrement: typeof VERSION_INCREMENT_OPTIONS[number];
+  tag: typeof TAG_OPTIONS[number];
+};
+
+type DistTags = {
+  [TAG.LATEST]: string;
+  [TAG.NEXT]: string;
+  [TAG.EXPERIMENTAL]: string;
+};
+
+type Tag = typeof TAG_OPTIONS[number];
+
+async function getGitBranch(): Promise<string> {
+  const { stdout } = await exec('git rev-parse --abbrev-ref HEAD');
+  return stdout.trim();
+}
+
+async function getDistTags(): Promise<DistTags> {
+  const { stdout } = await exec(`npm view ${packageName} dist-tags --json`);
+  const distTags = stdout.trim();
+  return JSON.parse(distTags) as DistTags;
+}
+
+async function cleanup(newVersion: string): Promise<void> {
+  await exec(`git reset HEAD~`);
+  await exec(`git checkout .`);
+  await exec(`git tag -d v${newVersion}`);
+}
+
+async function publish(tag: Tag): Promise<void> {
+  logger.log();
+  logger.log(`Publishing to ${tag}...`);
+  uiLine();
+  logger.log();
+
+  return new Promise((resolve, reject) => {
+    const childProcess = spawn('npm', ['publish', '--dry-run', '--tag', tag], {
```

Comment:
Just a reminder to take the dry run out of this before merging. Maybe it could be a supported flag?